pkg/apis/serving/v1alpha1/llm_inference_service_lifecycle.go

```diff
@@ -94,6 +94,7 @@ func (in *LLMInferenceService) DetermineWorkloadReadiness() {
 		in.GetStatus().GetCondition(PrefillWorkloadReady),
 		in.GetStatus().GetCondition(PrefillWorkerWorkloadReady),
 		in.GetStatus().GetCondition(SchedulerWorkloadReady),
+		in.GetStatus().GetCondition(RouterReady),
```

Comment:
I'd add a "GatewaysReady" condition and a "DetermineRouterReadiness", the `GatewaysReady` then gets aggregated to "RouterReady" condition similar to  `SchedulerWorkloadReady` for "WorkloadReady"